scripts/dns-any-query.nse

```diff
@@ -0,0 +1,159 @@
+local stdnse = require "stdnse"
+local dns = require "dns"
+local shortport = require "shortport"
+local target = require "target"
+
+description = [[
+Checks if ANY dns type is allowed.
+
+Usually if server allows ANY queries, it will send more than one answers.
+During prerule phase, if <code>newtargets</code> script argument is passed
+it will find new targets to nmap scanning queue
+]]
+
+---
+-- @usage
+-- nmap --script=dns-any-query -p53 <host>
+--
+-- Script uses dns library to connect to dns server
+--
+-- @output
+-- PORT   STATE SERVICE REASON
+-- 53/udp open  domain  udp-response ttl 55
+-- | dns-any-query:
+-- |   Result: Server is likely responding to ANY query
+-- |   Authoritative: TRUE
+-- |   Found:
+-- |     		MX
+-- |     		SOA
+-- |     		TXT
+-- |     		TXT
+-- |_    		TXT
+
+author = "Erhad Husovic"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"default","safe"}
+
+local options = {}
+
+-- During prerule phase, we except two variables: dns-any-query.server to query against and
+-- dns-any-query.query-domain domain to query
+-- Also, passing additional script arg <code>newtargets</code> will add domain name and server
+-- to new targets
+prerule = function()
+  options.domain_name, options.server, options.full_output,options.newtargets = stdnse.get_script_args(
+	"dns-any-query.query-domain","dns-any-query.server","dns-any-query.full-output","newtargets"
+  )
+
+  if not options.domain_name then
+    return false
+  end
+
+  if not options.server then
```

Comment:
i think it would be better to use `host.ip` if options.server is not passed

Reply:
Like in bellow comment, host is unknown in prerule phase

Reply:
ahh ok, however my main point was that `optons.domain_name` should be optional i.e. don't return false if it is not set.